src/snowflake/snowpark/dataframe.py

```diff
@@ -4188,6 +4189,117 @@ def _explain_string(self) -> str:
 
         return f"{msg}\n--------------------------------------------"
 
+    def map(
```

Comment:
A UDTF call may have other parameters like partitioning.

Reply:
As part of the previous comment, the`**kwargs` parameter can be used to pass arguments to the UDTF registration.

Reply:
@sfc-gh-lfallasavendano A more fundamental question here is why are we using UDTF to implement map. `.map` is supposed to give 1-1 input -> output map for each row. UDF is more aligned to do this.

Reply:
@sfc-gh-aalam  the reason for using UDTFs instead of UDFs is to support scenarios where non-scalar values are returned from the function that is being applied. For example:

```
df = session.create_dataframe([ [10,'a'], [20,'b'], [30,'c'] ], 
                              schema=StructType([StructField('N_ID', IntegerType()), StructField('NAME', StringType())]))

# Returning rows
df.map(lambda r: Row("---" + r.NAME + "---", r.N_ID), output_types=[StringType(), IntegerType()]).show()
-------------------
|"C_1"    |"C_2"  |
-------------------
|---c---  |30     |
|---a---  |10     |
|---b---  |20     |
-------------------


# Returning tuples:
def do_something(row):
    return ("---" + row.NAME + "---", row.N_ID)
df.map(do_something, output_types=[StringType(), IntegerType()]).show()
-------------------
|"C_1"    |"C_2"  |
-------------------
|---c---  |30     |
|---a---  |10     |
|---b---  |20     |
-------------------
```